### ISSUE TITLE
Question: RandomSearch will try max_trials times?

### ISSUE THREAD

**Issue by ZYHZ100** [keras-team/keras-tuner#192]
Hi,
If I set the max_trials to the value that is larger than the actual times where the search can run or that a combination of different parameters can lead to, then will the search stop before the max trial is reached? Thank you.

**Comment by ZYHZ100** [comment]
@omalleyt12 
Thank you so much

**Comment by omalleyt12** [comment]
@ZYHZ100 Thanks for the issue!

Yes, the search will stop. The stopping criterion right now is that RandomSearch will try and randomly generate hyperparameters combinations, and it will check if the generated combination has been tried before. If so, it will generate a new combination. If that happens 5 times in a row, it will stop the search.

So it's currently not guaranteed that the Tuner will try every single possible combination of hps before stopping. We're looking into ways of making the search exhaustive though

**Comment by rogeravs1997** [comment]
And whats the behaviour of BayeasianOptimization? I understand that the first "pair" of hyperparameters is chossen randomly and then every next "pair" is choosen with the goal to minimize the loss. Am i right? can you explain it?, please
